src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useState, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import { Error } from '../../types/Error';
+import { USER_ID, deleteTodo, updateTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  id: number;
+  title: string;
+  completed: boolean;
+  setErrorMessage: React.Dispatch<React.SetStateAction<Error>>;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  isInputLoading?: boolean;
+  deletingTodoIds: number[];
+};
+
+export const TodoItem: React.FC<Props> = ({
+  id,
+  title,
+  completed,
+  setErrorMessage,
+  setTodos,
+  isInputLoading = false,
+  deletingTodoIds,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isCompleted, setIsCompleted] = useState(completed);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTodoTitle, setEditedTodoTitle] = useState(title);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEditing) {
+      inputRef.current?.focus();
+    }
+  }, [isEditing]);
+
+  useEffect(() => {
+    setIsCompleted(completed);
+    setEditedTodoTitle(title);
+  }, [completed, title]);
+
+  const handleDeleteTodo = (todoId: number) => {
+    setIsLoading(true);
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage(Error.unableToDelete);
+        setTimeout(() => {
+          setErrorMessage(Error.none);
+        }, 3000);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  function handleToggleTodo(updatedTodo: Todo) {
+    setIsLoading(true);
+    updateTodo(updatedTodo)
+      .then(todo => {
+        setTodos(currentTodos => {
+          const newTodos = [...currentTodos];
+          const index = newTodos.findIndex(item => item.id === updatedTodo.id);
+
+          newTodos.splice(index, 1, todo);
+
+          return newTodos;
+        });
+        setIsCompleted(!isCompleted);
+      })
+      .catch(() => {
+        setErrorMessage(Error.unableToUpdate);
+        setTimeout(() => {
+          setErrorMessage(Error.none);
+        }, 3000);
+      })
+      .finally(() => setIsLoading(false));
+  }
+
+  function handleEditTodo(updatedTodo: Todo) {
```

Comment:
This comment still not fixed from the previous review, you need to fix it everywhere 
<img width="895" alt="image" src="https://github.com/user-attachments/assets/47ab8714-6224-4d5b-a299-8fbb1be2a022">
```suggestion
  const handleEditTodo = (updatedTodo: Todo) => {
```